drivers/mbox/mbox_andes_plic_sw.c

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright (c) 2022 Andes Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/mbox.h>
+
+#define LOG_LEVEL CONFIG_MBOX_LOG_LEVEL
+#include <zephyr/logging/log.h>
+#include <zephyr/irq.h>
+LOG_MODULE_REGISTER(mbox_andes_plic_sw);
+
+#define DT_DRV_COMPAT	andestech_plic_sw
+
+/* register definitions */
+#define PRIORITY_BASE 0x0
+#define PENDING_BASE  0x1000
+#define ENABLE_BASE   0x2000
+#define CLAIM_BASE    0x200004
+
+#define IRQ_REG(n) (n >> 5)
+
+#define PLIC_BASE(dev) \
+	((const struct mbox_andes_conf * const)(dev)->config)->base
+
+#define REG_PRIORITY(dev, irq)     (PLIC_BASE(dev) + PRIORITY_BASE + (irq << 2))
+#define REG_PENDING(dev, irq)      (PLIC_BASE(dev) + PENDING_BASE + \
+					(IRQ_REG(irq) << 2))
+#define REG_ENABLE(dev, hart, irq) (PLIC_BASE(dev) + ENABLE_BASE + \
+					(hart << 7) + IRQ_REG(irq))
+#define REG_CLAIM(dev, hart)       (PLIC_BASE(dev) + CLAIM_BASE + (hart << 12))
+
+#define IPI_NUM DT_INST_PROP(0, channel_max)
+#define IPI_MBOX_CHANNEL(node, prop, idx) MBOX_DT_CHANNEL_GET(node, ipi##idx)
```

Comment:
Please use a proper indentation for this.

Reply:
Thanks for your comments, I have update the source code.